File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/utilities/ConditionalMatch.java
package com.microsoft.recognizers.text.datetime.utilities;

import com.microsoft.recognizers.text.utilities.Match;

import java.util.Optional;

public class ConditionalMatch {

    private final Optional<Match> match;
    private final boolean success;

    public ConditionalMatch(Optional<Match> match, boolean success) {
        this.match = match;
        this.success = success;
    }

    public Optional<Match> getMatch() {

        return match;
    }

    public boolean getSuccess() {

        return success;
    }

}

File: .NET/Microsoft.Recognizers.Text.DateTime/Utilities/ConditionalMatch.cs
using System.Text.RegularExpressions;

namespace Microsoft.Recognizers.Text.DateTime
{
    public class ConditionalMatch
    {
        public ConditionalMatch(Match match, bool success)
        {
            Match = match;
            Success = success;
        }

        public Match Match { get; }

        public bool Success { get; }

        public int Index
        {
            get
            {
                return Match.Index;
            }
        }

        public int Length
        {
            get
            {
                return Match.Length;
            }
        }

        public string Value
        {
            get
            {
                return Match.Value;
            }
        }

        public GroupCollection Groups
        {
            get
            {
                return Match.Groups;
            }
        }
    }
}
